File: 1. Two Sum.js
// https://leetcode.com/problems/two-sum/

const nums = [0, 1, 0, 3, 12];
const target = 4;

function twoSum(nums, target) {
    var map = new Map();

    for (var i = 0; i < nums.length; i++) {
        if (map.has(target - nums[i])) {
            return [i, map.get(target - nums[i])];
        }
        map.set(nums[i], i);
    }

    return [];
}

console.log(twoSum(nums, target))

File: 1. Two Sum.kt

val target = 9;
val nums = listOf(0, 1, 0, 3, 12)

fun main(){  
    println(twoSum(nums, target))
}

fun twoSum(nums: List<Int>, target: Int): List<Int?> {
  var map  = HashMap<Int, Int> ()

  for ((index, value) in nums.withIndex()) {
    if(map.containsKey(target - value)){
      return listOf(index, map.get(target- value))
    }
      map.set(value, index);
  }  
  return listOf();
}
